importer/XkSq.py

```diff
@@ -1,62 +1,96 @@
-from Monument import Monument
+from Monument import Monument, Dataset
 import importer_utils as utils
+import importer as importer
 
 
 class XkSq(Monument):
 
-    def update_labels(self):
+    def set_is(self):
         """
-        Set the labels.
+        Set a more detailed value of P31.
 
-        TODO
-        Some of these contain several unprintable characters...
-        0093 - 0096
-        Maybe just go to the article and remove them?
-        https://sq.wikipedia.org/wiki/Lista_e_Monumenteve_n%C3%AB_Kosov%C3%AB
+        Based on 'category' and lookup table:
+        https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/xk_(sq)/category
+        If no match, default to default.
         """
-        print(self.name)
-        return
+        cat_dict = self.data_files["categories"]["mappings"]
+        monument_cat = self.category.lower()
+        lookup_match = utils.get_matching_items_from_dict(dict_name=cat_dict,
+                                                          value=monument_cat)
+        if len(lookup_match) == 1:
+            self.add_statement("is", lookup_match[0])
+        else:
+            super().set_is()
 
-    def set_no(self):
-        self.add_statement("kosovo_monument_id", str(self.idno))
+    def set_inception(self):
+        """Set inception if it's a whole year."""
+        if utils.legit_year(self.date_period):
+            inc_year = {"time_value": {"year": self.date_period}}
+            self.add_statement("inception", inc_year)
+        else:
+            # this will give a lot of results... :/
+            self.add_to_report("date_period", self.date_period, "inception")
+
+    def set_location(self):
+        """Set Location if it's matched against valid settlements."""
+        if self.has_non_empty_attribute("place"):
+            settl_dic = self.data_files["settlements"]
+            s_match = utils.get_item_from_dict_by_key(dict_name=settl_dic,
+                                                      search_term=self.place,
+                                                      search_in="itemLabel")
+            if len(s_match) == 1:
+                self.add_statement("location", s_match[0])
+            else:
+                self.add_to_report("place", self.place, "location")
 
     def set_adm_location(self):
-        """
-        Set the administrative location.
+        """Set Admin Location if it's matched against valid values."""
+        municip_raw = self.municipality.lower()
+        municp_dic = self.data_files["municipalities"]["mappings"]
+        municip_matches = utils.get_matching_items_from_dict(
+            municip_raw, municp_dic)
+        if len(municip_matches) == 1:
+            self.add_statement("located_adm", municip_matches[0])
+        else:
+            self.add_to_report(
+                "municipality", self.municipality, "located_adm")
 
-        The values are never wikilinked.
-        E panjohur = unknown
```

Comment:
This line would still be nice to have (since I encountered it in the previews)

Reply:
Just spotted this in the online mapping. Maybe set it to "-" there to make it visible. Then no need to keep it here.